<<<<<<< HEAD:Waterfall-Proxy-Patches/0019-Don-t-declare-uuid-unless-it-s-null.patch
From d9cb38b627c2493dc55f4306301719f7e3301605 Mon Sep 17 00:00:00 2001
=======
From ae64b8938eadb7ee4b0c68572b4e270530603ad9 Mon Sep 17 00:00:00 2001
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0018-Don-t-declare-uuid-unless-it-s-null.patch
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Wed, 16 Dec 2020 18:43:17 +0800
Subject: [PATCH] Don't declare uuid unless it's null


diff --git a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
<<<<<<< HEAD:Waterfall-Proxy-Patches/0019-Don-t-declare-uuid-unless-it-s-null.patch
index 1cebd8bc..aa952717 100644
=======
index e643d8a7..dc5b64b7 100644
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0018-Don-t-declare-uuid-unless-it-s-null.patch
--- a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
@@ -490,7 +490,10 @@ public class InitialHandler extends PacketHandler implements PendingConnection
                     {
                         loginProfile = obj;
                         name = obj.getName();
-                        uniqueId = Util.getUUID( obj.getId() );
+                        // FlameCord - Don't declare uuid unless it's null
+                        if (uniqueId == null) {
+                            uniqueId = Util.getUUID(obj.getId());
+                        }
                         finish();
                         return;
                     }
-- 
2.32.0.windows.1

